src/ci_workflow/ci_check_gradle_dependencies.py

```diff
@@ -23,13 +23,17 @@ def __init__(self, component: InputComponent, git_repo: GitRepository, target: C
 
     def __get_dependencies(self) -> PropertiesFile:
         cmd = " ".join(
-            [
-                f"./gradlew {self.gradle_project or ''}:dependencies",
-                f"-Dopensearch.version={self.target.opensearch_version}",
-                f"-Dbuild.snapshot={str(self.target.snapshot).lower()}",
-                "--configuration compileOnly",
-                '| grep -e "---"',
-            ]
+            filter(
```

Comment:
what does filter do here?

Reply:
I see. Thanks! I guess that what I was missing in my branch.

Reply:
Just wondering how @dblock made it work as per this conversation: https://github.com/opensearch-project/opensearch-build/pull/1731/files#r823034371

Reply:
This is similar to how I add distribution:
https://github.com/dblock/opensearch-build/blob/8babdde32e2262b9e2de3e3672daa198bef74c4f/src/build_workflow/builder_from_source.py#L45